Update permissions & some remnants from docs (#11)
discord-userdoccers/discord-userdoccers: pages/topics/permissions.mdx

# Permissions

Permissions are a way to limit and grant certain abilities to users in Discord. A set of base permissions can be configured at the guild level for different roles. When these roles are attached to users, they grant or revoke specific privileges within the guild. Along with the guild-level permissions, Discord also supports permission overwrites that can be assigned to individual roles or members on a per-channel basis.

> > info
> > [Application command permissions](/interactions/application-commands#permissions) allow you to enable or disable specific commands for entire channels in addition to individual roles or users.

Reviewer: Info boxes aren't done like this anymore, see CONTRIBUTING.md

```suggestion
<Alert type="info">

[Application command permissions](/interactions/application-commands#permissions) allow you to enable or disable specific commands for entire channels in addition to individual roles or users.

</Alert>
```

Permissions are stored in a variable-length integer serialized into a string, and are calculated using bitwise operations. For example, the permission value `123` will be serialized as `"123"`. For long-term stability, it's recommended to deserialize the permissions using your preferred languages' Big Integer libraries. The total permissions integer can be determined by OR-ing (`|`) together each individual value, and flags can be checked using AND (`&`) operations.

In API v8 and above, all permissions are serialized as strings, including the `allow` and `deny` fields in overwrites. Any new permissions are rolled back into the base field.

> > info
> > In [API v6 (now deprecated)](/reference#api-versions), the `permissions`, `allow`, and `deny` fields in roles and overwrites are still serialized as a number; however, these numbers shall not grow beyond 31 bits. During the remaining lifetime of API v6, all new permission bits will only be introduced in `permissions_new`, `allow_new`, and `deny_new`. These `_new` fields are just for response serialization; requests with these fields should continue to use the original `permissions`, `allow`, and `deny` fields, which accept both string or number values.

Reviewer: Same here.

```suggestion
<Alert type="info">

In [API v7 and below (now deprecated)](/reference#api-versions), the `permissions`, `allow`, and `deny` fields in roles and overwrites are still serialized as a number; however, these numbers shall not grow beyond 31 bits. During the remaining lifetime of these API versions, all new permission bits will only be introduced in `permissions_new`, `allow_new`, and `deny_new`. These `_new` fields are just for response serialization; requests with these fields should continue to use the original `permissions`, `allow`, and `deny` fields, which accept both string or number values.

</Alert>
```

###### Bitwise Permission Flags

| Permission                             | Value                            | Description                                                                                                                                         | Channel Type |
| -------------------------------------- | -------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------- | ------------ |
| CREATE_INSTANT_INVITE                  | `0x0000000000000001` `(1 << 0)`  | Allows creation of instant invites                                                                                                                  | T, V, S      |

> | KICK_MEMBERS \*                        | `0x0000000000000002` `(1 << 1)`  | Allows kicking members                                                                                                                              |              |
> | BAN_MEMBERS \*                         | `0x0000000000000004` `(1 << 2)`  | Allows banning members                                                                                                                              |              |
> | ADMINISTRATOR \*                       | `0x0000000000000008` `(1 << 3)`  | Allows all permissions and bypasses channel permission overwrites                                                                                   |              |
> | MANAGE_CHANNELS \*                     | `0x0000000000000010` `(1 << 4)`  | Allows management and editing of channels                                                                                                           | T, V, S      |
> | MANAGE_GUILD \*                        | `0x0000000000000020` `(1 << 5)`  | Allows management and editing of the guild                                                                                                          |              |

Reviewer: No spaces after asterisks.

```suggestion
| KICK_MEMBERS\*                         | `0x0000000000000002` `(1 << 1)`  | Allows kicking members                                                                                                                              |              |
| BAN_MEMBERS\*                          | `0x0000000000000004` `(1 << 2)`  | Allows banning members                                                                                                                              |              |
| ADMINISTRATOR\*                        | `0x0000000000000008` `(1 << 3)`  | Allows all permissions and bypasses channel permission overwrites                                                                                   |              |
| MANAGE_CHANNELS\*                      | `0x0000000000000010` `(1 << 4)`  | Allows management and editing of channels                                                                                                           | T, V, S      |
| MANAGE_GUILD\*                         | `0x0000000000000020` `(1 << 5)`  | Allows management and editing of the guild                                                                                                          |              |
```

| STREAM                                 | `0x0000000000000200` `(1 << 9)`  | Allows the user to go live                                                                                                                          | V            |
| VIEW_CHANNEL                           | `0x0000000000000400` `(1 << 10)` | Allows guild members to view a channel, which includes reading messages in text channels and joining voice channels                                 | T, V, S      |
| SEND_MESSAGES                          | `0x0000000000000800` `(1 << 11)` | Allows for sending messages in a channel and creating threads in a forum (does not allow sending messages in threads)                               | T, V         |
| SEND_TTS_MESSAGES                      | `0x0000000000001000` `(1 << 12)` | Allows for sending of `/tts` messages                                                                                                               | T, V         |

> | MANAGE_MESSAGES \*                     | `0x0000000000002000` `(1 << 13)` | Allows for deletion of other users messages                                                                                                         | T, V         |

Reviewer: Same here.

```suggestion
| MANAGE_MESSAGES\*                      | `0x0000000000002000` `(1 << 13)` | Allows for deletion of other users messages                                                                                                         | T, V         |
```

| MOVE_MEMBERS                           | `0x0000000001000000` `(1 << 24)` | Allows for moving of members between voice channels                                                                                                 | V, S         |
| USE_VAD                                | `0x0000000002000000` `(1 << 25)` | Allows for using voice-activity-detection in a voice channel                                                                                        | V            |
| CHANGE_NICKNAME                        | `0x0000000004000000` `(1 << 26)` | Allows for modification of own nickname                                                                                                             |              |
| MANAGE_NICKNAMES                       | `0x0000000008000000` `(1 << 27)` | Allows for modification of other users nicknames                                                                                                    |              |

> | MANAGE_ROLES \*                        | `0x0000000010000000` `(1 << 28)` | Allows management and editing of roles                                                                                                              | T, V, S      |
> | MANAGE_WEBHOOKS \*                     | `0x0000000020000000` `(1 << 29)` | Allows management and editing of webhooks                                                                                                           | T, V         |
> | MANAGE_EXPRESSIONS \*                  | `0x0000000040000000` `(1 << 30)` | Allows management and editing of emojis, stickers and soundboard sounds                                                                             |              |

Reviewer: Same here.

```suggestion
| MANAGE_ROLES\*                         | `0x0000000010000000` `(1 << 28)` | Allows management and editing of roles                                                                                                              | T, V, S      |
| MANAGE_WEBHOOKS\*                      | `0x0000000020000000` `(1 << 29)` | Allows management and editing of webhooks                                                                                                           | T, V         |
| MANAGE_EXPRESSIONS\*                   | `0x0000000040000000` `(1 << 30)` | Allows management and editing of emojis, stickers and soundboard sounds                                                                             |              |
```

> | USE_APPLICATION_COMMANDS               | `0x0000000080000000` `(1 << 31)` | Allows members to use application commands, including slash commands and context menu commands.                                                     | T, V         |
> | REQUEST_TO_SPEAK                       | `0x0000000100000000` `(1 << 32)` | Allows for requesting to speak in stage channels.                                                                                                   | S            |

Reviewer: No periods.

```suggestion
| USE_APPLICATION_COMMANDS               | `0x0000000080000000` `(1 << 31)` | Allows members to use application commands, including slash commands and context menu commands                                                      | T, V         |
| REQUEST_TO_SPEAK                       | `0x0000000100000000` `(1 << 32)` | Allows for requesting to speak in stage channels                                                                                                    | S            |
```

| MANAGE_EVENTS                          | `0x0000000200000000` `(1 << 33)` | Allows for creating, editing, and deleting scheduled events                                                                                         | V, S         |

> | MANAGE_THREADS \*                      | `0x0000000400000000` `(1 << 34)` | Allows for deleting and archiving threads, and viewing all private threads                                                                          | T            |

Reviewer: Same here.

```suggestion
| MANAGE_THREADS\*                       | `0x0000000400000000` `(1 << 34)` | Allows for deleting and archiving threads, and viewing all private threads                                                                          | T            |
```

| CREATE_PRIVATE_THREADS                 | `0x0000001000000000` `(1 << 36)` | Allows for creating private threads                                                                                                                 | T            |
| USE_EXTERNAL_STICKERS                  | `0x0000002000000000` `(1 << 37)` | Allows the usage of custom stickers from other servers                                                                                              | T, V         |
| SEND_MESSAGES_IN_THREADS               | `0x0000004000000000` `(1 << 38)` | Allows for sending messages in threads                                                                                                              | T            |
| USE_EMBEDDED_ACTIVITIES                | `0x0000008000000000` `(1 << 39)` | Allows for using Activities (applications with the `EMBEDDED` flag) in a voice channel                                                              | V            |

> | MODERATE_MEMBERS \*\*                  | `0x0000010000000000` `(1 << 40)` | Allows for timing out users to prevent them from sending or reacting to messages in chat and threads, and from speaking in voice and stage channels |              |

Reviewer: Same here.

```suggestion
| MODERATE_MEMBERS\*\*                   | `0x0000010000000000` `(1 << 40)` | Allows for timing out users to prevent them from sending or reacting to messages in chat and threads, and from speaking in voice and stage channels |              |
```

> | VIEW_CREATOR_MONETIZATION_ANALYTICS \* | `0x0000020000000000` `(1 << 41)` | Allows for viewing guild role subscriptions insights                                                                                                |              |

Reviewer: Same here.

```suggestion
| VIEW_CREATOR_MONETIZATION_ANALYTICS\*  | `0x0000020000000000` `(1 << 41)` | Allows for viewing guild role subscriptions insights                                                                                                |              |
```

**\* These permissions require the owner account to use [two-factor authentication](/topics/oauth2#twofactor-authentication-requirement) when used on a guild that has server-wide 2FA enabled.**

> **\*\* See [Permissions for Timed Out Members](/topics/permissions#permissions-for-timed-out-members) to understand how permissions are temporarily modified for timed out users.**

Reviewer: These shouldn't be bolded.

> Note that permission names may be referred to differently in the Discord client. For example, "Manage Permissions" refers to MANAGE_ROLES, "Use Voice Activity" refers to USE_VAD, and "Timeout Members" refers to MODERATE_MEMBERS.

Reviewer: Permission names should be surrounded by ``.

```suggestion
Note that permission names may be referred to differently in the Discord client. For example, "Manage Permissions" refers to `MANAGE_ROLES`, "Use Voice Activity" refers to `USE_VAD`, and "Timeout Members" refers to `MODERATE_MEMBERS`.
```

Users must have the `VIEW_CHANNEL` permission to view _any_ threads in the channel, even if they are directly mentioned or added to the thread.

## Permission Syncing

Permissions with regards to categories and channels within categories are a bit tricky. Rather than inheritance, permissions are calculated by means of what we call Permission Syncing. If a child channel has the same permissions and overwrites (or lack thereof) as its parent category, the channel is considered "synced" to the category. Any further changes to a **parent category** will be reflected in its synced child channels. Any further changes to a **child channel** will cause it to become de-synced from its parent category, and its permissions will no longer change with changes to its parent category.

### Role Object

Roles represent a set of permissions attached to a group of users. Roles have names, colors, and can be "pinned" to the side bar, causing their members to be listed separately. Roles can have separate permission profiles for the global context (guild) and channel context. The `@everyone` role has the same ID as the guild it belongs to.

###### Role Structure

> | Field          | Type                                                                    | Description                                      |
> | -------------- | ----------------------------------------------------------------------- | ------------------------------------------------ |
> | id             | snowflake                                                               | Role id                                          |
> | name           | string                                                                  | Role name                                        |
> | description    | string?                                                                 | The description for the role (max 90 characters) |
> | color          | integer                                                                 | Integer representation of hexadecimal color code |
> | hoist          | boolean                                                                 | If this role is pinned in the user listing       |
> | icon?          | string?                                                                 | Role [icon hash](/reference#image-formatting)    |
> | unicode_emoji? | string?                                                                 | Role unicode emoji                               |
> | position       | integer                                                                 | Position of this role                            |
> | permissions    | string                                                                  | Permission bit set                               |
> | managed        | boolean                                                                 | Whether this role is managed by an integration   |
> | mentionable    | boolean                                                                 | Whether this role is mentionable                 |
> | tags?          | [role tags](/topics/permissions#role-object-role-tags-structure) object | The tags this role has                           |

Reviewer: ```suggestion
| Field          | Type                                                                    | Description                                      |
| -------------- | ----------------------------------------------------------------------- | ------------------------------------------------ |
| id             | snowflake                                                               | The ID of the role                                          |
| name           | string                                                                  | The name of the role                                        |
| description    | string?                                                                 | The description for the role (max 90 characters) |
| color          | integer                                                                 | Integer representation of a hexadecimal color code for the role |
| hoist          | boolean                                                                 | Whether this role is pinned in the user listing       |
| icon?          | string?                                                                 | The role's [icon hash](/reference#image-formatting)    |
| unicode_emoji? | string?                                                                 | The role's unicode emoji                               |
| position       | integer                                                                 | Position of this role                            |
| permissions    | string                                                                  | Permission bit set                               |
| managed        | boolean                                                                 | Whether this role is managed by an integration   |
| mentionable    | boolean                                                                 | Whether this role is mentionable                 |
| tags?          | [role tags](/topics/permissions#role-object-role-tags-structure) object | The tags this role has                           |
```

Roles without colors (`color == 0`) do not count towards the final computed color in the user list.

###### Role Tags Structure

Tags with type `null` represent booleans. They will be present and set to `null` if they are "true", and will be not present if they are "false".

| Field                    | Type      | Description                                        |
| ------------------------ | --------- | -------------------------------------------------- |
| bot_id?                  | snowflake | The ID of the bot this role belongs to             |
| integration_id?          | snowflake | The ID of the integration this role belongs to     |

> | premium_subscriber?      | null      | Whether this is the guild's Booster role           |

Reviewer: ```suggestion
| premium_subscriber?      | null      | Whether this is the guild's premium subscriber (booster) role           |
```

> | subscription_listing_id? | snowflake | The ID of this role's subscription sku and listing |

Reviewer: ```suggestion
| subscription_listing_id? | snowflake | The ID of this role's subscription SKU and listing |
```
